core/src/main/java/hudson/slaves/SlaveComputer.java

```diff
@@ -992,6 +999,18 @@ public String call() throws IOException {
             catch (Throwable ex) { return "< 1.335"; } // Older agent.jar won't have VERSION
         }
     }
+
+    private static final class CommunicationProtocol extends MasterToSlaveCallable<String,IOException> {
+        @Override
+        public String call() throws IOException {
+            try {
+                if (Engine.current() != null)
+                    return Engine.current().getProtocolName();
+                return Launcher.getCommunicationProtocolName();
+            } catch (Throwable ex) { return null; } // remote not support this feature
```

Comment:
This code will catch `Error`s which are expected to be fatal in the most of cases. Is it intended?

Reply:
I referred to the other part of the code, but not good at least for this change. I'll fix it.